File: examples/node/chat-completion.js
import 'dotenv/config';
import OpenAI from 'openai';

const client = new OpenAI({
  apiKey: process.env.INFERGATE_API_KEY,
  baseURL: process.env.INFERGATE_BASE_URL || 'https://api.useinfergate.com/v1',
});

const model = process.env.INFERGATE_MODEL || 'gpt-5.5';

const response = await client.chat.completions.create({
  model,
  messages: [
    { role: 'system', content: 'You are a concise assistant.' },
    { role: 'user', content: 'Say hello from InferGate in one sentence.' },
  ],
});

console.log(response.choices[0].message.content);

File: examples/python/chat_completion.py
import os
from dotenv import load_dotenv
from openai import OpenAI

load_dotenv()

client = OpenAI(
    api_key=os.getenv("INFERGATE_API_KEY"),
    base_url=os.getenv("INFERGATE_BASE_URL", "https://api.useinfergate.com/v1"),
)

model = os.getenv("INFERGATE_MODEL", "gpt-5.5")

response = client.chat.completions.create(
    model=model,
    messages=[
        {"role": "system", "content": "You are a concise assistant."},
        {"role": "user", "content": "Say hello from InferGate in one sentence."},
    ],
)

print(response.choices[0].message.content)
